len8/checker.py

```diff
@@ -97,18 +115,48 @@ def exclude(self, excludes: t.List[Path]) -> None:
         self._exclude = excludes
 
     @property
-    def extend(self) -> bool:
-        """If ``True``, increase acceptable line length to 99.
+    def extend(self) -> int:
+        """The extension factor.
 
         Returns:
-            ``bool``
+            ``int``
         """
         return self._extend
 
     @extend.setter
-    def extend(self, extend: bool) -> None:
+    def extend(self, extend: int) -> None:
+        if not 0 <= extend <= 2:
+            raise ValueError("'extend' should be between 0 and 2 inclusive")
         self._extend = extend
 
+    @property
+    def code_length(self) -> int:
+        """The value to use as the maximum line length for code. This
+        will return:
+
+        - The custom length, if :obj:`_code_length` is not ``None``.
```

Comment:
I dont think predicating what this returns on a private attribute is a good idea. Maybe something like "if a custom length was set" or something.